app/views/incidents/show.html.haml

```diff
@@ -27,12 +27,14 @@
     %p
       %strong= human_name :incident, :root_cause_analysis
       = simple_format @incident.root_cause_analysis
-    %p
-      %strong= human_name :incident, :reason_code
-      = @incident.reason_code.full_label
-    %p
-      %strong= human_name :incident, :supplementary_reason_code
-      = @incident.supplementary_reason_code.full_label
+    - if @incident.reason_code_id?
```

Comment:
At this point, all completed, valid incidents will have `reason_code` and a `supplementary_reason_code`. You can probably lose these two conditionals.